tests/data/dataset_readers/wikitables/wikitables_dataset_reader_test.py

```diff
@@ -11,17 +12,244 @@ def test_reader_reads(self):
+                "<e,d> -> [<e,<e,d>>, e]",
+                "<e,d> -> number",
+                "<e,e> -> ['lambda x', e]",
+                "<e,e> -> [<#1,<#1,#1>>, e]",
+                "<e,e> -> [<<#1,#2>,<#2,#1>>, <e,e>]",
+                "<e,p> -> ['lambda x', p]",
+                "<e,p> -> [<<#1,#2>,<#2,#1>>, <p,e>]",
+                "<e,r> -> ['lambda x', r]",
+                "<e,r> -> [<<#1,#2>,<#2,#1>>, <r,e>]",
+
+                # These are instance-specific production rules.  These are the columns in the
+                # table.
+                # TODO(mattg): this also seems backwards.
+                "<e,r> -> fb:row.row.avg_attendance",
+                "<e,r> -> fb:row.row.division",
+                "<e,r> -> fb:row.row.league",
+                "<e,r> -> fb:row.row.open_cup",
+                "<e,r> -> fb:row.row.playoffs",
+                "<e,r> -> fb:row.row.regular_season",
+                "<e,r> -> fb:row.row.year",
+
+                # PART_TYPE rules.
+                # TODO(mattg): what is a cell part?
+                "<p,<<d,p>,p>> -> [<d,<p,<<d,p>,p>>>, d]",
+                "<p,d> -> ['lambda x', d]",
+                "<p,d> -> [<<#1,#2>,<#2,#1>>, <d,p>]",
+                "<p,e> -> ['lambda x', e]",
+                "<p,e> -> [<<#1,#2>,<#2,#1>>, <e,p>]",
+                "<p,e> -> fb:cell.cell.part",
+                "<p,p> -> ['lambda x', p]",
+                "<p,p> -> [<#1,<#1,#1>>, p]",
+                "<p,p> -> [<<#1,#2>,<#2,#1>>, <p,p>]",
+                "<p,r> -> ['lambda x', r]",
+                "<p,r> -> [<<#1,#2>,<#2,#1>>, <r,p>]",
+
+                # Functions that operate on rows.
+                "<r,<<d,r>,r>> -> [<d,<r,<<d,r>,r>>>, d]",
+                "<r,d> -> ['lambda x', d]",
+                "<r,d> -> [<<#1,#2>,<#2,#1>>, <d,r>]",
+                "<r,e> -> ['lambda x', e]",
+                "<r,e> -> [<<#1,#2>,<#2,#1>>, <e,r>]",
+                "<r,p> -> ['lambda x', p]",
+                "<r,p> -> [<<#1,#2>,<#2,#1>>, <p,r>]",
+                "<r,r> -> ['lambda x', r]",
+                "<r,r> -> [<#1,<#1,#1>>, r]",
+                "<r,r> -> [<<#1,#2>,<#2,#1>>, <r,r>]",
+                "<r,r> -> fb:row.row.next",
+
+                # All valid start types - we just say that any BASIC_TYPE is a valid start type.
+                "@START@ -> d",
+                "@START@ -> e",
+                "@START@ -> p",
+                "@START@ -> r",
+
+                # Now we get to productions for the basic types.  The first part of each section
+                # here has rules for expanding the basic type with function applications; after
+                # that we get to terminal productions (and a "BASIC_TYPE -> x" rule for lambda
+                # variable productions).
+
+                # Date and numbers.  We don't have any terminal productions for these, as for some
+                # reason terminal numbers are actually represented as CELL_TYPE.
```

Comment:
NLTK assigns `EntityType` to all constants by default, and I let that be the case for numbers. However, I also defined `CELL_TYPE` to be the same as `EntityType` for no good reason. I think I should change the latter.

Reply:
I added a TODO for this.